### PR TITLE
[Ready] Show databases for users with schema access in sqllab

### LINKED ISSUE #8081: Unable to grant per datasource/schema access with 0.34.0rc1
Original issue.
https://github.com/apache/incubator-superset/issues/7554

You can't use SQL Lab if you don't grant `database` permission to user. It makes superset unusable with `per datasource` access model. You can still see dashboards, but you can't manually query any data.

If you add `database` permission, user can access all tables in database which is not acceptable.

### Expected results

See all available database/schema/table in SQL Lab based on granted datasource access.

### Actual results

SQL Lab dropdowns are empty.

#### How to reproduce the bug

- create user
- grant it only datasource permission without database access.

### PR DESCRIPTION
**Pull request by bkyryliuk**
### CATEGORY

Choose one

- [X] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [X] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

This change enables user that have sql lab access and have access to some DB schemas to use sql lab.
Partially fixes: https://github.com/apache/incubator-superset/issues/8081

### TEST PLAN

* added unit test
* tested in the dev environment & logging in as different users

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/incubator-superset/issues/8081
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/security_tests.py [hunk 1]
```diff
@@ -28,8 +30,46 @@ def get_perm_tuples(role_name):
...
+            "schema_access", ds.schema_perm
+        )
+        security_manager.add_permission_role(
+            security_manager.find_role(SCHEMA_ACCESS_ROLE), schema_perm_view
+        )
+        gamma_user = security_manager.find_user(username="gamma")
+        gamma_user.roles.append(security_manager.find_role(SCHEMA_ACCESS_ROLE))
+        session.commit()
+
+        GET_SQL_DBS_REQUEST = "databaseasync/api/read?_flt_0_expose_in_sqllab=1&" \
```
Comment (dpgaspar) [inline tests/security_tests.py]: Endpoint should be: `/api/v1/database/?q=(keys:!(none),filters:!((col:expose_in_sqllab,opr:eq,value:!t)),order_columns:database_name,order_direction:asc,page:0,page_size:-1)`

**Comment by bkyryliuk** [comment]
@dpgaspar - added that url as well, will keep both for now in tests.
The one I have gotten - was from the app network requests.

Tests are fixed.

File: tests/security_tests.py [hunk 2]
```diff
@@ -28,8 +30,65 @@ def get_perm_tuples(role_name):
...
+            "_oc_DatabaseAsync=database_name&_od_DatabaseAsync=asc"
+        )
+        self.login(username="gamma")
+        databases_json = self.client.get(OLD_FLASK_GET_SQL_DBS_REQUEST).json
+        assert databases_json["count"] == 1
+
+        NEW_FLASK_GET_SQL_DBS_REQUEST = (
+            "/api/v1/database/?q=(keys:!(none),filters:!((col:expose_in_sqllab,opr:eq,value:!t)),"
+            "order_columns:database_name,order_direction:asc,page:0,page_size:-1)"
+        )
```
Comment (dpgaspar) [inline tests/security_tests.py]: To make the code more readable we could take advantage of Rison and user something like:
``` python
arguments = {
    "keys": ["none"],
    "filters":[
        {
            "col": "expose_in_lab",
            "opr": "eq",
            "value": true
        }
    ],
    "order_columns": "database_name"
    "order_direction": "asc"
    "page": 0
    "page_size": -1
}
NEW_FLASK_GET_SQL_DBS_REQUEST = f"/api/v1/database/?q={prison.dumps(arguments)}"
```

File: superset/assets/package.json [hunk 3]
```diff
@@ -93,6 +93,7 @@
     "chrono-node": "^1.3.11",
     "classnames": "^2.2.5",
     "core-js": "^3.2.1",
+    "cypress": "^3.4.1",
```
Comment (dpgaspar) [inline superset/assets/package.json]: Do we really need this change, is it related to the PR?

**Comment by dpgaspar** [comment]
No sure about the change made on `DatabaseFilter` 

@mistercrunch

**Comment by bkyryliuk** [comment]
@dpgaspar thanks for the comments, resolved those

**Pull request closed without merging**